src/producer/mod.rs

```diff
@@ -1,5 +1,9 @@
 use crate::error::Error;
-use crate::proto::{self, Client, Info, Job, Push, QueueAction, QueueControl};
+use crate::proto::{
+    self, parse_provided_or_from_env, Client, Info, Job, Push, QueueAction, QueueControl,
+};
+#[cfg(feature = "ent")]
+use crate::proto::{Batch, BatchHandle, CommitBatch, OpenBatch};
```

Comment:
The `cfg`-gated `use` should probably be in their own section, otherwise the sorting gets a little weird.